### PR TITLE
Allow square brackets in external issue patterns

### PR DESCRIPTION
**Pull request by derkoe**
Many projects write the issue reference in square brackets, like `[JIRA-123]` or `[#123]`. This PR makes the external issue pattern support this as well.

### CODE REVIEW

File: modules/markup/html_test.go [hunk 1]
```diff
@@ -342,6 +342,7 @@ func TestRegExp_IssueNumericPattern(t *testing.T) {
 		"#1234",
 		"#0",
 		"#1234567890987654321",
+		"[#1234]",
 	}
 	falseTestCases := []string{
 		"# 1234",
```
Comment (sapk) [inline modules/markup/html_test.go:351]: Just to make sure, I would add to falseTestCases `[]` `[X]` that are checklist element even if there should be replace before by markdown.

**Review by sapk — APPROVED** [review]
Thanks for your PR.  
The CI fail seems not related. 
You can restart it via a commit amend or maybe a owner can restart the build.

**Comment by lafriks** [comment]
Doesn't this interfere with links for example, `[PR-121](http://example.com)` ?

File: modules/markup/html_test.go (hunk 1 above)
Comment (derkoe) [reply modules/markup/html_test.go:351]: Added the test cases

**Comment by derkoe** [comment]
Links are handled before the issue patterns in `RenderSpecialLink` - I think this should not interfere

**Comment by thehowl** [comment]
LGTM

**Pull request merged**